### PR TITLE
Expand testing to system/integration test with known result

### PR DESCRIPTION
**Pull request by jkitzes**
Quick update to the testing rule:
1. Rather than a smoke test, which would simply tell if the program runs at all (which will be obvious to a novice who tries to run the code on real data), I'd suggest a single system/integration test for the whole pipeline, ensuring you get at least one known correct result for a known input (this might have been the original intent of the line anyway)
2. I moved the testing rule from Collaboration to Software

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -166,7 +166,9 @@ Regarding metadata, [Elizabeth Wickes][wickes-comment-metadata] writes:
 7.  Commenting/uncommenting are not routinely used to control program behavior
     *   Use if/else to control behavior
     *   Use configuration files or command-line arguments for parameters
-8.  Submit code to a reputable DOI-issuing repository upon submission of paper, just like data
+8.  And a simple example or test data set to run to tell if it's working at all and whether it gives a known correct output for a simple known input
```
Comment (lexnederbragt) [inline index.md]: And --> Use

File: index.md [hunk 2]
```diff
@@ -166,7 +166,9 @@ Regarding metadata, [Elizabeth Wickes][wickes-comment-metadata] writes:
...
+    *   A system/integration test that checks that the entire program at once for a case similar to the real analysis
```
Comment (lexnederbragt) [inline index.md]: checks that the entire program --> checks the entire program

**Comment by lexnederbragt** [comment]
You talk about "the whole pipeline" in the comment above, but about "the entire program at once" in the PR. Do you mean to separate the two?

Not entirely sure about the move to the 'Software' section, but perhaps others can chime in?

**Comment by jkitzes** [comment]
Thanks Lex, fixed the typos. I thought that having testing in the Collaboration section was a typo - it seems closer to the other bullets in Software to me. But I could be convinced otherwise -

**Comment by jennybc** [comment]
I use somewhat vague tests or assertions in these settings. Examples: test that the output has certain dimensions or at least `n` rows or all values are positive, etc. I agree it is valuable to give concrete examples of what sorts of tests are appropriate here, since it's not a classic unit test.

I also think the move from collaboration to software is sensible. (However, the one tie to collaboration is that I like to run these tests after a collaborator has had their hands all over things, to make sure nothing broke.)

**Pull request merged**